src/device.ts

```diff
@@ -61,6 +61,94 @@ export class Device {
     this.id = `${id}`;
   }
 
+  getId(): string {
+    return this.id;
+  }
+
+  getContext(): string {
+    return this['@context'];
+  }
+
+  setContext(context: string): void {
+    this['@context'] = context;
+  }
+
+  getTypes(): string[] {
+    return this['@type'];
+  }
+
+  setTypes(type: string[]): void {
+    this['@type'] = type;
+  }
+
+  addType(type: string): void {
+    this['@type'].push(type);
```

Comment:
It would be really helpful if this method could de-duplicate types and ensure that the new one doesn't already exist.

Reply:
Done